### DISCUSSION TITLE
zmap formula's patch for JSON support actually removes JSON support flag

### DISCUSSION THREAD

**Discussion by manuelbua** [Homebrew/discussions#3178] [Everyday usage] [answered]
The `zmap` formula contains a patch to fix `json-c 0.14` compatibility during compilation (https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/zmap.rb#L59).
The problem with this is that the patch, which successfully replace the `;` with ` `inside the `JSON_FLAGS` value, also removes the `-DJSON` flag definition that actually defines JSON support.
```patch
-    add_definitions("-DJSON")
+    string(REPLACE ";" " " JSON_CFLAGS "${JSON_CFLAGS}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${JSON_CFLAGS}")
```

It should instead read:
```patch
     add_definitions("-DJSON")
+    string(REPLACE ";" " " JSON_CFLAGS "${JSON_CFLAGS}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${JSON_CFLAGS}")
```

I'm not sure how to propose such a patch for the formula as i'm not familiar with writing formulas, so i opened this discussion.

**Comment by SMillerDev** [comment, marked as answer]
I created https://github.com/Homebrew/homebrew-core/pull/99167 to fix this

**Reply by manuelbua** [reply]
Thanks @SMillerDev!
